packages/terra-dropdown-button/src/DropdownButton.jsx

```diff
@@ -57,29 +57,57 @@ class DropdownButton extends React.Component {
     this.handleDropdownRequestClose = this.handleDropdownRequestClose.bind(this);
     this.handleKeyDown = this.handleKeyDown.bind(this);
     this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.setButtonNode = this.setButtonNode.bind(this);
+    this.getButtonNode = this.getButtonNode.bind(this);
+    this.setListNode = this.setListNode.bind(this);
+    this.state = { isOpen: false, isActive: false, isKeyboardEvent: false };
+  }
+
+  setListNode(element) {
+    this.dropdownList = element;
+  }
 
-    this.state = { isOpen: false, isActive: false };
+  setButtonNode(node) {
+    this.buttonNode = node;
+  }
+
+  getButtonNode() {
+    return this.buttonNode;
   }
 
   handleDropdownButtonClick() {
+    if (this.state.isOpen) {
+      this.setState({ isKeyboardEvent: false });
+    }
     this.setState(prevState => ({ isOpen: !prevState.isOpen }));
   }
 
   handleDropdownRequestClose(callback) {
     this.setState({ isOpen: false }, typeof callback === 'function' ? callback : undefined);
+    this.setState({ isKeyboardEvent: false, isActive: false });
```

Comment:
These set state calls can probably be combined. React should batch state updates, but I am not familiar with what happens when some of the batched calls have a callback function. The callback can probably be moved into a variable to shorten the declaration.

Reply:
changed  : [here](https://github.com/cerner/terra-core/blob/cf4a1d2744dbd30b29ba636528a1ebd6c739c3a0/packages/terra-dropdown-button/src/DropdownButton.jsx#L86-L87)